godot-core/src/registry/class.rs

```diff
@@ -337,6 +352,39 @@ pub(crate) fn try_dynify_object<T: GodotClass, D: ?Sized + 'static>(
     Err(error.into_error(object))
 }
 
+pub(crate) fn get_dyn_property_hint_string<D>() -> GString
+where
+    D: ?Sized + 'static,
+{
+    let typeid = any::TypeId::of::<D>();
+    let dyn_traits_by_typeid = global_dyn_traits_by_typeid();
+    let Some(relations) = dyn_traits_by_typeid.get(&typeid) else {
+        let trait_name = sys::short_type_name::<D>();
+        godot_warn!("No gdext class has been linked to trait {trait_name} with #[godot_dyn].");
+        return GString::default();
+    };
+
+    let mut relations = relations.iter();
+    let first_relation = relations
+        .next()
+        .map(|r| r.implementing_class_name.to_string())
+        .unwrap_or_else(|| {
+            panic!(
+                "\
+    Trait {trait_name} has been registered as DynGd Trait \
+    despite no class being related to it \
+    **this is a bug, please report it** \
+    ",
+                trait_name = sys::short_type_name::<D>()
+            )
+        });
+
+    GString::from(relations.fold(
+        first_relation,
+        |hint_string, relation| format! {"{hint_string},{r}", r=relation.implementing_class_name},
+    ))
```

Comment:
Instead of "calculate initial value + `fold()`", would [`reduce()`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.reduce) be a possibility? Generally it's not so nice that the `implementing_class_name` needs to be extracted twice.

I'm also OK with [keeping it simple](https://en.wikipedia.org/wiki/KISS_principle#In_software_development) and just repeatedly appending to a string in a `for` loop. 

When I write code myself, if functional approaches can't be done in an elegant and obvious way, I usually prefer procedural ones. Simple-to-understand code is better than clever one 🙂

Reply:
i think we could do something like
```rs
let relations: Vec<String> = relations.iter().map(|rel| rel.implementing_class_name.to_string()).collect();
let hint_string: String = relations.join(",");
// check if is empty and error 
```

Reply:
It's not great that the easiest way here is an inefficient one (intermediate `Vec` allocated, on top of all temporary strings). Not that it matters much, but even the `for` loop doesn't have this problem.

I now remember this wasn't the first time this annoyed me, we can probably just use this:
https://github.com/godot-rust/gdext/blob/6a5d19d5f8e7563730ccd4738e46f398726b222a/godot-ffi/src/toolbox.rs#L165-L181

You could even call this on the outer collection and pass the extractor function as `format_elem` parameter.

Reply:
I replaced fold monstrosity (applied together with better format of check if there are any relations for given `DynGd`) with `join_with`